coordinator/statement_executor.go

```diff
@@ -56,10 +56,10 @@ type StatementExecutor struct {
 }
 
 // ExecuteStatement executes the given statement with the given execution context.
-func (e *StatementExecutor) ExecuteStatement(stmt influxql.Statement, ctx query.ExecutionContext) error {
+func (e *StatementExecutor) ExecuteStatement(stmt influxql.Statement, ctx *query.ExecutionContext) error {
 	// Select statements are handled separately so that they can be streamed.
 	if stmt, ok := stmt.(*influxql.SelectStatement); ok {
-		return e.executeSelectStatement(context.Background(), stmt, &ctx)
+		return e.executeSelectStatement(context.Background(), stmt, ctx)
```

Comment:
This is confusing as we now have two independent `context.Context` instances. What is the reasoning behind embedding it within `ExecutionContext` and leaving it as an argument?

The longer term goal is that the `context.Context` would be passed in via the `QueryExecutor#Execute` method, thus coming from the HTTP request or whoever initiated the query.

Reply:
That appears to be a mistake. The intention is to remove the first argument and use the third (now second) as the first argument.